plugins/outputs/cloudwatchlogs/pusher_test.go

```diff
@@ -767,6 +769,184 @@ func TestResendWouldStopAfterExhaustedRetries(t *testing.T) {
+			expectedStrategy: retryShort,
+		},
+		"ServiceUnavailableException": {
+			err:              &cloudwatchlogs.ServiceUnavailableException{},
+			expectedStrategy: retryLong,
+		},
+		"ThrottlingException": {
+			err:              &cloudwatchlogs.ThrottlingException{},
+			expectedStrategy: retryLong,
+		},
+		"500 - InternalFailure": {
+			err:              awserr.NewRequestFailure(awserr.New("500", "InternalFailure", nil), 500, ""),
+			expectedStrategy: retryLong,
+		},
+		"503 - ServiceUnavailable": {
+			err:              awserr.NewRequestFailure(awserr.New("503", "ServiceUnavailable", nil), 503, ""),
+			expectedStrategy: retryLong,
+		},
+		"Connection Refused": {
+			err:              syscall.ECONNREFUSED,
+			expectedStrategy: retryLong,
+		},
+		"Connection Reset By Peer": {
+			err:              syscall.ECONNRESET,
+			expectedStrategy: retryLong,
+		},
+		"Connection Timeout": {
+			err:              syscall.ETIMEDOUT,
+			expectedStrategy: retryLong,
+		},
+		"Deadline Exceeded": {
+			err:              os.ErrDeadlineExceeded,
+			expectedStrategy: retryLong,
+		},
+		"Other Errors": {
+			err:              errors.New("Unknown Error"),
+			expectedStrategy: retryShort,
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			chosen := chooseRetryWaitStrategy(tt.err)
+			require.Equal(t, tt.expectedStrategy, chosen)
+		})
+	}
+}
+
+func TestRetryWaitShort(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		n           int
+		minDuration time.Duration
+		maxDuration time.Duration
+	}{
+		{
+			n:           0,
+			minDuration: 100 * time.Millisecond,
+			maxDuration: 200 * time.Millisecond,
+		},
+		{
+			n:           1,
+			minDuration: 200 * time.Millisecond,
+			maxDuration: 400 * time.Millisecond,
+		},
+		{
+			n:           2,
+			minDuration: 400 * time.Millisecond,
+			maxDuration: 800 * time.Millisecond,
+		},
+		{
+			n:           3,
+			minDuration: 800 * time.Millisecond,
+			maxDuration: 1600 * time.Millisecond,
+		},
+		{
+			n:           4,
+			minDuration: 1600 * time.Millisecond,
+			maxDuration: 3200 * time.Millisecond,
+		},
+		{
+			n:           5,
+			minDuration: 30 * time.Second,
+			maxDuration: 1 * time.Minute,
+		},
+		{
+			n:           6,
+			minDuration: 30 * time.Second,
+			maxDuration: 1 * time.Minute,
+		},
+		{
+			n:           7,
+			minDuration: 30 * time.Second,
+			maxDuration: 1 * time.Minute,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(fmt.Sprintf("%d", tt.n), func(t *testing.T) {
+			for _ = range 1000 {
+				duration := retryWaitShort(tt.n)
+				assert.GreaterOrEqual(t, duration, tt.minDuration, "retryWait(%v) should be greater than or equal to %v", tt.n, tt.minDuration)
+				assert.LessOrEqual(t, duration, tt.maxDuration, "retryWait(%v) should be less than or equal to %v", tt.n, tt.maxDuration)
+			}
+		})
+	}
+}
+
+func TestRetryWaitLong(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		n           int
```

Comment:
nit: Don't need n since `range` will give you the index.
```go
for n, tt := range tests
```

Reply:
That's true, but I think it's more clear when the test case specifies the test conditions rather than relying on the index to form the test case inputs.

Reply:
Yeah you're right since they're separate test cases.